crates/bindings-csharp/Runtime/Internal/Module.cs

```diff
@@ -252,15 +252,33 @@ private static byte[] Consume(this BytesSource source)
                 // but a module should be prepared for it.
                 case 0:
                     break;
+                case (short)(ushort)FFI.Errno.NO_SUCH_BYTES:
+                    throw new NoSuchBytesException();
                 default:
-                    throw new UnreachableException();
+                    throw new UnknownException((FFI.Errno)(ushort)ret);
+            }
+        }
+    }
+
+    private static void Write(this BytesSink sink, byte[] bytes)
+    {
+        var start = 0U;
+        while (true)
+        {
+            var written = (uint)bytes.Length;
+            var buffer = bytes.AsSpan((int)start);
+            FFI._bytes_sink_write(sink, buffer, ref written);
+            start += written;
+            if (start == bytes.Length)
```

Comment:
Nit: just change the condition to `while (start != bytes.Length)` instead of short-circuiting?